### PR TITLE
Enforce JSX inline conditional as a ternary

### LINKED ISSUE #1979: prohibit `&&` conditional rendering in the return statement of a component
```jsx
// avoid
const WrapTingy = ({ cond }) => (
  cond && <Tingy/>
);
 
// that would error if cond is `undefined`, use this version instead:
 
const WrapTingy = ({ cond }) => (
  <>{cond && <Tingy/>}</>
);
```

### PR DESCRIPTION
**Pull request by holic**
This fixes a fairly common bug where React will render the falsy left side of a conditional with `&&` operator, resulting in things like `0` showing up in your UI. See https://github.com/jsx-eslint/eslint-plugin-react/pull/2888#issuecomment-751413618.

Fixes #1979
Closes #2888 

cc @ljharb

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -124,16 +125,15 @@ module.exports = {
...
-      ],
+      plugins: ['react'],
       parserOptions: {
         ecmaFeatures: {
           jsx: true,
         },
       },
       rules: {
         'react/display-name': 2,
+        'react/jsx-inline-conditional': 2,
```
Comment (holic) [inline index.js:136]: Because this is helps identify+fix a certain class of rendering bugs and isn't strictly stylistic, I enabled this by default. Let me know if I should change this!

Comment (ljharb) [reply index.js:136]: Adding a rule to the recommended config is a breaking change, so we'll basically never be doing that. Please remove it.

File: index.js [hunk 2]
```diff
@@ -158,9 +158,7 @@ module.exports = {
       },
     },
     all: {
-      plugins: [
-        'react',
-      ],
+      plugins: ['react'],
```
Comment (ljharb) [inline index.js:161]: please revert unrelated formatting changes; if perhaps you're running prettier on repos that do not use it, please don't do that either :-)

File: tests/lib/rules/jsx-inline-conditional.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+      },
+    },
+    {
+      code: '<div>{possiblyNull ?? <SomeComponent />}</div>',
+      parser: parsers.TYPESCRIPT_ESLINT,
+    },
+    {
+      code: '<div>{possiblyNull ?? <SomeComponent />}</div>',
+      parser: parsers['@TYPESCRIPT_ESLINT'],
+    },
```
Comment (ljharb) [inline tests/lib/rules/jsx-inline-conditional.js:53]: these should not be needed, since the parsers.all helper should automatically run every test case in every parser.
```suggestion
```

File: tests/lib/rules/jsx-inline-conditional.js [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+      code: '<div>{someCondition ? <div>{anotherCondition ? <SomeComponent /> : null}</div> : null}</div>',
+    },
+    {
+      code: '<div>{someCondition && someOtherCondition ? <SomeComponent /> : null}</div>',
+    },
+    {
+      code: '<div>{possiblyNull ?? <SomeComponent />}</div>',
+      parserOptions: {
+        ecmaVersion: 2020,
+      },
```
Comment (ljharb) [inline tests/lib/rules/jsx-inline-conditional.js:44]:
```suggestion
      code: '<div>{possiblyNull ?? <SomeComponent />}</div>',
      features: ['nullish coalescing'],
```
and then we'll need to add that feature to the `parsers.all` helper

**Review by ljharb — CHANGES_REQUESTED** [review]
Doesn't this duplicate the recently added `jsx-no-leaked-render` rule?

**Comment by ljharb** [comment]
I think #3203 may have covered this?
